packages/rate-limit/rate-limit.js

```diff
@@ -133,7 +143,10 @@ RateLimiter.prototype.check = function (input) {
   var reply = {
     allowed: true,
     timeToReset: 0,
-    numInvocationsLeft: Infinity
+    numInvocationsLeft: Infinity,
+    userId: input.userId,
+    clientAddress: input.clientAddress,
+    connectionId: input.connectionId,
```

Comment:
Any reason to not return the entire `input`?  It seems like `type` (e.g. `method` or `publication`) and `name` (Of the publication or method) are still left out here and they would be very useful.

Perhaps it might be better to just pass the input to `_executeCallback` separately? (`_executeCallback(reply, ruleInput)`)?

Also, the JSDoc comment above needs to be changed to match whatever we decide.